File: 15-18/Master1/app/src/main/java/com/wx/master1/MainActivity.java
package com.wx.master1;

import android.databinding.ViewDataBinding;
import android.view.View;

import com.wx.master1.activity.JsonActivity;
import com.wx.master1.base.BaseActivity;
import com.wx.master1.databinding.ActivityMainBinding;

public class MainActivity extends BaseActivity {

    private ActivityMainBinding binding;

    @Override
    public int getLayoutId() {
        return R.layout.activity_main;
    }

    @Override
    public void setDataBinding(ViewDataBinding binding) {
        this.binding = (ActivityMainBinding) binding;
    }

    @Override
    public void initData() {
        binding.btnJson.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                openActivity(JsonActivity.class);
            }
        });
    }



}

File: Android-Demo/UiMaster/app/src/main/java/com/wx/uimaster/MainActivity.kt
package com.wx.uimaster

import android.databinding.ViewDataBinding
import android.view.View
import com.wx.uimaster.activity.ComplexRecyclerViewActivity

import com.wx.uimaster.base.BaseActivity
import com.wx.uimaster.databinding.ActivityMainBinding

class MainActivity : BaseActivity(), View.OnClickListener {

    private lateinit var binding: ActivityMainBinding

    override fun layoutId(): Int {
        return R.layout.activity_main
    }

    override fun setDataBinding(binding: ViewDataBinding) {
        this.binding = binding as ActivityMainBinding
    }

    override fun initData() {
        binding.complexRecyclerview.setOnClickListener(this)
    }

    override fun onClick(view: View?) {
        when(view?.id) {
            R.id.complex_recyclerview -> openActivity(ComplexRecyclerViewActivity::class.java)
        }
    }

}
